pkg/controller/azure/cache/claim.go

```diff
@@ -68,24 +66,34 @@ func ConfigureRedis(_ context.Context, cm resource.Claim, cs resource.Class, mg
 		return errors.Errorf("expected resource claim %s to be %s", cm.GetName(), cachev1alpha1.RedisClusterGroupVersionKind)
 	}
 
-	rs, csok := cs.(*corev1alpha1.ResourceClass)
+	rs, csok := cs.(*v1alpha1.RedisClass)
 	if !csok {
-		return errors.Errorf("expected resource class %s to be %s", cs.GetName(), corev1alpha1.ResourceClassGroupVersionKind)
+		return errors.Errorf("expected resource class %s to be %s", cs.GetName(), v1alpha1.RedisClassGroupVersionKind)
 	}
 
 	i, mgok := mg.(*v1alpha1.Redis)
 	if !mgok {
 		return errors.Errorf("expected managed resource %s to be %s", mg.GetName(), v1alpha1.RedisGroupVersionKind)
 	}
 
-	spec := v1alpha1.NewRedisSpec(rs.Parameters)
+	spec := &v1alpha1.RedisSpec{
+		ResourceSpec: corev1alpha1.ResourceSpec{
+			ReclaimPolicy: corev1alpha1.ReclaimRetain,
+		},
+		RedisParameters: rs.SpecTemplate.RedisParameters,
+	}
 	if err := resolveAzureClassValues(rc); err != nil {
 		return errors.Wrap(err, "cannot resolve Azure class instance values")
 	}
 
+	// NOTE(hasheddan): consider moving defaulting to either CRD or managed reconciler level
+	if spec.RedisConfiguration == nil {
+		spec.RedisConfiguration = map[string]string{}
```

Comment:
is it necessary here to instantiate the map? does something break downstream if you leave it as nil?

Reply:
As a general rule with any of this "defaulting" behavior I added in any of the claim controllers, I added it to satisfy the claim_tests. What was happening before is the `Parse___()` function for each kind was setting these values to empty types (i.e. `map[string]string{}` here), but setting the `spec` to the `specTemplate` does not do the same. I did some light investigating into how these values were being used down the line (i.e. how they are submitted to the specific cloud provider API) and I believe it will be possible to omit some of this behavior. However, in the interest of efficiency and safety, I opted for just keeping the behavior as close to existing as possible. These notes are for whoever is responsible of ensuring high fidelity for each cloud provider to call out addressing this. I think we may want to look into CRD defaulting behavior, which was [introduced in Kubernetes v1.15](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#defaulting).

Reply:
Got it, that makes sense to have this defaulting to stay aligned with the previous behavior in the old `Parse___()` functions.